utils/src/main/scala/za/co/absa/enceladus/utils/validation/SchemaValidator.scala

```diff
@@ -44,6 +46,36 @@ object SchemaValidator {
     errorsAccumulator.toList
   }
 
+  /**
+    * Validates the error column.
+    * Most of the time the error columns should not exist in the input schema. But uf it does exist, it should
+    * conform to the expected type.
+    * Nullability of the error column is not an issue.
+    *
+    * @param schema A Spark schema
+    * @return A list of ValidationErrors objects, each containing a column name and the list of errors and warnings
+    */
+  def validateErrorColumn(schema: StructType)
+                         (implicit spark: SparkSession)
+                         : List[FieldValidationFailure] = {
+    val expectedTypeNonNullable = ArrayType.apply(ErrorMessage.errorColSchema, containsNull = false)
+    val expectedTypeNullable = ArrayType.apply(ErrorMessage.errorColSchema, containsNull = true)
+    val errCol = schema.find(f => f.name == ErrorMessage.errorColumnName)
+    errCol match {
+      case Some(errColField) =>
+        if (errColField.dataType != expectedTypeNonNullable && errColField.dataType != expectedTypeNullable) {
+          val actualType = errColField.dataType
+          List(FieldValidationFailure(errColField.name, "",
+            ValidationError("The error column in the input data does not conform to the expected type. " +
+              s"Expected: $expectedTypeNonNullable, actual: $actualType") :: Nil))
```

Comment:
Wouldn't we rather for them to see the nullable type as that is the one that we create.

Reply:
I'm not 100% sure, need to check. But I think we create a non-nullable error column field, but when it is saved to Parquet and then reopened for a second Standardization/Conformance, the field becomes nullable.